tools/mo/openvino/tools/mo/middle/SSliceComplex.py

```diff
@@ -0,0 +1,127 @@
+            ],
+            edges=[
+                ('strided_slice_real', 'strided_slice_real_data'),
+                ('strided_slice_imag', 'strided_slice_imag_data'),
+                ('strided_slice_real_data', 'complex', {'in': 0}),
+                ('strided_slice_imag_data', 'complex', {'in': 1}),
+            ])
+
+    def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
+        strided_slice_real = match['strided_slice_real']
+        strided_slice_imag = match['strided_slice_imag']
+
+        real_input = strided_slice_real.in_port(0).get_source().node
+        imag_input = strided_slice_imag.in_port(0).get_source().node
+        if real_input.id != imag_input.id:
+            log.debug('The pattern does not correspond to operation for complex tensor. Different inputs.')
+            return
+
+        real_slices = np.array(strided_slice_real.slices)
+        imag_slices = np.array(strided_slice_imag.slices)
+
+        zeros_in_real_input_slices = np.argwhere(real_slices==0).flatten()
+        ones_in_imag_input_slices = np.argwhere(imag_slices==1).flatten()
+
+        if len(zeros_in_real_input_slices) != 1 or len(ones_in_imag_input_slices) != 1:
+            return
+
+        slice_dim_for_real_part = zeros_in_real_input_slices[0]
+        slice_dim_for_imag_part = ones_in_imag_input_slices[0]
+        if slice_dim_for_real_part != slice_dim_for_imag_part:
+            return
+
+        emulated_complex_tensor_shape = strided_slice_real.in_port(0).data.get_shape()
+        if emulated_complex_tensor_shape is None:
+            return
+
+        emulated_complex_tensor_rank = len(emulated_complex_tensor_shape)
+        complex_node = match['complex']
+
+        for dst in complex_node.out_port(0).get_connection().get_destinations():
+            after_complex_node = dst.node
+            after_complex_node['input_rank_changed'] = True
+
+        input_slices_have_ellipsis = len(np.argwhere(real_slices == Ellipsis).flatten()) != 0
+
+        if slice_dim_for_real_part == emulated_complex_tensor_rank - 1 or input_slices_have_ellipsis:
+            complex_node.out_port(0).get_connection().set_source(strided_slice_real.in_port(0).get_source())
+        else:
+            complex_node_name = complex_node.soft_get('name', complex_node.id)
+            perm = int64_array([*range(0, slice_dim_for_real_part),
+                                *range(slice_dim_for_real_part + 1, emulated_complex_tensor_rank),
+                                slice_dim_for_real_part])
+            transpose = create_op_with_const_inputs(graph, Transpose, {1: perm},
+                                                    {'name': complex_node_name + '/cmplx'})
+            complex_node.out_port(0).get_connection().set_source(transpose.out_port(0))
+            strided_slice_real.in_port(0).get_source().connect(transpose.in_port(0))
+            rename_nodes([(complex_node, complex_node_name + '/to_be_removed'), (transpose, complex_node_name)])
```

Comment:
New line

Reply:
let me change it during next MO change ;)